### PR TITLE
chore: Adding explicit bind to redis and sentinel for IPv4 clusters #5957

### LINKED ISSUE #5957: Redis/sentinel not working with IPv4 cluster
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

I upgraded to ArgoCD v2.0.0-rc2 and the `argocd-redis-ha-server` stateful set wasn't starting. The logs contained address family errors.

Additionally, `argocd-server` is unable to connect to the `argocd-redis-ha-haproxy`.

**To Reproduce**

Deploy ArgoCD v2.0.0-rc2 to a Kubernetes cluster with IPv6 disabled. 

**Expected behavior**

Redis and sentinel start up cleanly. 

**Version**

```shell
❯ argocd version                                                                                                     
argocd: v1.8.7+eb3d1fb.dirty
  BuildDate: 2021-03-07T19:33:57Z
  GitCommit: eb3d1fb84b9b77cdffd70b14c4f949f1c64a9416
  GitTreeState: dirty
  GoVersion: go1.16
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v2.0.0-rc2+9603ae3
  BuildDate: 2021-03-29T21:29:13Z
  GitCommit: 9603ae37765dadd6e6db519896b8065ca277a775
  GitTreeState: clean
  GoVersion: go1.16
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: v3.9.4 2021-02-09T19:22:10Z
  Helm Version: v3.5.1+g32c2223
  Kubectl Version: v0.20.4
  Jsonnet Version: v0.17.0
```

**Logs**
argocd-server
```
redis: 2021/04/03 20:33:44 pubsub.go:168: redis: discarding bad PubSub connection: EOF
redis: 2021/04/03 20:33:44 pubsub.go:168: redis: discarding bad PubSub connection: EOF
redis: 2021/04/03 20:33:44 pubsub.go:168: redis: discarding bad PubSub connection: EOF
redis: 2021/04/03 20:33:44 pubsub.go:168: redis: discarding bad PubSub connection: write tcp 172.29.68.15:43426->10.100.200.54:6379: write: broken pipe
redis: 2021/04/03 20:33:44 pubsub.go:168: redis: discarding bad PubSub connection: write tcp 172.29.68.15:43426->10.100.200.54:6379: write: broken pipe
redis: 2021/04/03 20:33:44 pubsub.go:168: redis: discarding bad PubSub connection: EOF
redis: 2021/04/03 20:33:44 pubsub.go:168: redis: discarding bad PubSub connection: EOF
redis: 2021/04/03 20:33:44 pubsub.go:168: redis: discarding bad PubSub connection: EOF
redis: 2021/04/03 20:33:44 pubsub.go:168: redis: discarding bad PubSub connection: EOF
```
argocd-redis-ha-server
```
13:C 03 Apr 2021 19:53:15.706 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
13:C 03 Apr 2021 19:53:15.706 # Redis version=6.2.1, bits=64, commit=00000000, modified=0, pid=13, just started
13:C 03 Apr 2021 19:53:15.706 # Configuration loaded
13:M 03 Apr 2021 19:53:15.706 * monotonic clock: POSIX clock_gettime
13:M 03 Apr 2021 19:53:16.003 # Could not create server TCP listening socket ::*:6379: unable to bind socket, errno: 97
```
argocd-redis-ha-haproxy
```
[NOTICE] 092/202813 (1) : New worker #1 (7) forked
[WARNING] 092/202821 (7) : Server check_if_redis_is_master_0/R0 is DOWN, reason: Layer4 timeout, info: " at step 1 of tcp-check (connect)", check duration: 3000ms. 2 active and 0 backup servers left. 0 sessions active, 0 requeued, 0 remaining in queue.
[WARNING] 092/202821 (7) : Server check_if_redis_is_master_0/R1 is DOWN, reason: Layer4 timeout, info: " at step 1 of tcp-check (connect)", check duration: 3000ms. 1 active and 0 backup servers left. 0 sessions active, 0 requeued, 0 remaining in queue.
[WARNING] 092/202821 (7) : Server check_if_redis_is_master_0/R2 is DOWN, reason: Layer4 timeout, info: " at step 1 of tcp-check (connect)", check duration: 3000ms. 0 active and 0 backup servers left. 0 sessions active, 0 requeued, 0 remaining in queue.
[ALERT] 092/202821 (7) : backend 'check_if_redis_is_master_0' has no server available!
[WARNING] 092/202821 (7) : Server check_if_redis_is_master_1/R0 is DOWN, reason: Layer4 timeout, info: " at step 1 of tcp-check (connect)", check duration: 3000ms. 2 active and 0 backup servers left. 0 sessions active, 0 requeued, 0 remaining in queue.
[WARNING] 092/202821 (7) : Server check_if_redis_is_master_1/R1 is DOWN, reason: Layer4 timeout, info: " at step 1 of tcp-check (connect)", check duration: 3000ms. 1 active and 0 backup servers left. 0 sessions active, 0 requeued, 0 remaining in queue.
[WARNING] 092/202821 (7) : Server check_if_redis_is_master_1/R2 is DOWN, reason: Layer4 timeout, info: " at step 1 of tcp-check (connect)", check duration: 3000ms. 0 active and 0 backup servers left. 0 sessions active, 0 requeued, 0 remaining in queue.
[ALERT] 092/202821 (7) : backend 'check_if_redis_is_master_1' has no server available!
[WARNING] 092/202821 (7) : Server check_if_redis_is_master_2/R0 is DOWN, reason: Layer4 timeout, info: " at step 1 of tcp-check (connect)", check duration: 3000ms. 2 active and 0 backup servers left. 0 sessions active, 0 requeued, 0 remaining in queue.
[WARNING] 092/202821 (7) : Server check_if_redis_is_master_2/R1 is DOWN, reason: Layer4 timeout, info: " at step 1 of tcp-check (connect)", check duration: 3000ms. 1 active and 0 backup servers left. 0 sessions active, 0 requeued, 0 remaining in queue.
[WARNING] 092/202821 (7) : Server check_if_redis_is_master_2/R2 is DOWN, reason: Layer4 timeout, info: " at step 1 of tcp-check (connect)", check duration: 3000ms. 0 active and 0 backup servers left. 0 sessions active, 0 requeued, 0 remaining in queue.
[ALERT] 092/202821 (7) : backend 'check_if_redis_is_master_2' has no server available!
[WARNING] 092/202821 (7) : Server bk_redis_master/R0 is DOWN, reason: Layer4 timeout, info: " at step 1 of tcp-check (connect)", check duration: 3000ms. 2 active and 0 backup servers left. 4 sessions active, 0 requeued, 0 remaining in queue.
[WARNING] 092/202821 (7) : Server bk_redis_master/R1 is DOWN, reason: Layer4 timeout, info: " at step 1 of tcp-check (connect)", check duration: 3000ms. 1 active and 0 backup servers left. 0 sessions active, 0 requeued, 0 remaining in queue.
[WARNING] 092/202821 (7) : Server bk_redis_master/R2 is DOWN, reason: Layer4 timeout, info: " at step 1 of tcp-check (connect)", check duration: 3000ms. 0 active and 0 backup servers left. 0 sessions active, 0 requeued, 0 remaining in queue.
[ALERT] 092/202821 (7) : backend 'bk_redis_master' has no server available!
```

I was able to get redis and sentinel running by adding the following to the redis.conf and sentinel.conf
```
bind 127.0.0.1
```

I am still trying to figure out what is happening with `argocd-redis-ha-haproxy`.

### PR DESCRIPTION
**Pull request by rumstead**
Fixes #5957

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: manifests/ha/base/redis-ha/generate.sh [hunk 1]
```diff
@@ -14,3 +14,4 @@ helm template argocd ./chart \
 sed -e 's/check inter 1s/check inter 3s/' ./chart/upstream_orig.yaml >> ./chart/upstream.yaml && rm ./chart/upstream_orig.yaml
 sed -i.bak 's/timeout server 30s/timeout server 6m/' ./chart/upstream.yaml && rm ./chart/upstream.yaml.bak
 sed -i.bak 's/timeout client 30s/timeout client 6m/' ./chart/upstream.yaml && rm ./chart/upstream.yaml.bak
+sed -i.bak -E 's/^([[:space:]]){8}sentinel replaceme argocd/    bind/' ./chart/upstream.yaml && rm ./chart/upstream.yaml.bak
```
Comment (rumstead) [inline manifests/ha/base/redis-ha/generate.sh:17]: I am not proud of this but I am not sure of a better way. The templating doesn't support just [adding "raw" config values](https://github.com/DandyDeveloper/charts/blob/5a93b9545a4791321cf3762ff85338afa119672c/charts/redis-ha/templates/_configs.tpl#L59) like Redis.

Comment (alexmt) [reply manifests/ha/base/redis-ha/generate.sh:17]: It works :)

**Review by alexmt — CHANGES_REQUESTED** [review]
Thank you @rumstead ,

Can you please refresh generated manifests (`make manifiests`).

Or jsut merge this https://github.com/rumstead/argo-cd/pull/1  - I had to regenerate manifests to test the change on my cluster.

**Comment by rumstead** [comment]
> Thank you @rumstead ,
> 
> Can you please refresh generated manifests (`make manifiests`).
> 
> Or jsut merge this [rumstead#1](https://github.com/rumstead/argo-cd/pull/1) - I had to regenerate manifests to test the change on my cluster.

Done!

**Review by alexmt — APPROVED** [review]
LGTM!

**Pull request merged**